src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -803,15 +802,16 @@ void updateInCache
 
   if (mscInsert == 0)  // 0: Insertion was really made
   {
-    subCacheUpdateStatisticsIncrement();
-
-    if (subCacheP != NULL)
+    subCacheStatisticsIncrementUpdates();
+  // FIXME PR: I understand this block is not needed with std::map based
+  // logic
+  /*  if (subCacheP != NULL)
     {
       LM_T(LmtSubCache, ("Calling subCacheItemRemove"));
       subCacheItemRemove(subCacheP);
     }
+   */
```

Comment:
I think this part needs to be as it was.

subCacheP comes from a lookup in the cache (a call to subCacheItemLookup) and it can perfectly be NULL, which only means that the subscription didn't exist in the cache (new creation).

If it didn't previously exist, it cannot be removed.
subCacheItemRemove may even crash the system is called with a NULL pointer ...